### DISCUSSION TITLE
Problems self-hosting on Debian

### DISCUSSION THREAD

**Discussion by Jerroder** [ente-io/ente#4433] [Q&A]
### Description

Hi,

I've been trying to follow [the doc](https://github.com/ente-io/ente/blob/main/server/RUNNING.md#run-without-docker) to get it running without docker, but it hardly make sense.

So I installed go, postgres, libsodium and pkg-config. Then created the database, ran it, and create the user `postgres`, that's the easy part

1. Now in `Start museum`, I need to set the variable `ENTE_DB_USER` to `postgres` but if I use air for live reloads, the user is `ente_user`? Are these supposed to be different and not the same one?

2. For `go run cmd/museum/main.go` I had to figure out that I had to clone the repo as well as find `main.go` and then `cd` into `server` so that `cmd/museum/main.go` is right. When I ran this, it downloaded a bunch of stuff and then exited without any error. Is this expected? I would guess main.go is the main program but it doesn't run anything other than downloads. When I run it again, I only get those messages (air gives the same):

```
package command-line-arguments
        imports github.com/ente-io/museum/pkg/api
        imports github.com/ente-io/museum/pkg/controller
        imports github.com/ente-io/museum/pkg/controller/access
        imports github.com/ente-io/museum/pkg/repo
        imports github.com/ente-io/museum/pkg/utils/crypto
        imports github.com/GoKillers/libsodium-go/cryptogenerichash: build constraints exclude all Go files in /home/xxx/go/pkg/mod/github.com/!go!killers/libsodium-go@v0.0.0-20171022220152-dd733721c3cb/cryptogenerichash
package command-line-arguments
        imports github.com/ente-io/museum/pkg/api
        imports github.com/ente-io/museum/pkg/controller
        imports github.com/ente-io/museum/pkg/controller/access
        imports github.com/ente-io/museum/pkg/repo
        imports github.com/ente-io/museum/pkg/utils/crypto
        imports github.com/GoKillers/libsodium-go/cryptosecretbox: build constraints exclude all Go files in /home/xxx/go/pkg/mod/github.com/!go!killers/libsodium-go@v0.0.0-20171022220152-dd733721c3cb/cryptosecretbox
```

3. Do I need to fill `museum.yaml` with DB user and password? If so, where do I get the password? And `museum.yaml` should be created "in the current directory" but where is that? `ente/`? `ente/server/`?

4. If it works eventually, how do I stop, start and reload it?

### What product are you using?

Ente Photos

### What platform are you using?

Server

**Comment by mngshm** [comment]
Please check: https://github.com/ente-io/ente/discussions/3901. I moved this issue to the discussions, because as of now we are not dealing with problems in self hosting ente through github issues. 

Thank You for your cooperation.
